### ISSUE TITLE
Variable substitution for files or dirs must be cross-platform

### ISSUE THREAD

**Issue by kaste** [SublimeLinter/SublimeLinter#1265]
That's easier with an example. If the plugin author states

```
	"config": "${folder}/packages.json"
```

this must resolve to 

```
	"/foo/bar/packages.json"
OR: "C:\foo\bar\packages.json"
```

Maybe it is better to add a trailing slash to the vars. And then: `"${folder}packages.json"`

**Comment by FichteFoll** [comment]
Well, Windows translates forward slashes to backslashes in most of their API methods and it's quite hard to determine which setting or which part of a setting is supposed to represent a path and adjust any slashes occuring. The `config` setting isn't a standard setting, or at least it doesn't occur anywhere in the code.

Adding a trailing slash is sensible and usually safe because multiple slashes are collapsed, but ST itself doesn't provide this for its variables either, so imo it's fine to leave them out.

```py
>>> window.extract_variables()['folder']
'/home/fichte/code/SublimeLinter3'
>>> window.extract_variables()['file_path']
'/home/fichte/code/SublimeLinter3/lint'
```

**Comment by kaste** [comment]
Windows is the mother of all compatibility layers. You're right. If I exec 

```
py -3 -m flake8 "C:\Users\c-flo\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter/lint/backend.py"
```

Windows does not complain and just runs away.
